cmd/kubeadm/app/cmd/init.go

```diff
@@ -72,7 +71,7 @@ var (
 		You can now join any number of machines by running the following on each node
 		as root:
 
-		  kubeadm join --token {{.Token}} {{.MasterIP}}:{{.MasterPort}} --discovery-token-ca-cert-hash {{.CAPubKeyPin}}
+		  kubeadm join --token {{.Token}} {{.MasterEndpoint}} --discovery-token-ca-cert-hash {{.CAPubKeyPin}}
```

Comment:
With this change the join string displayed by `kubeadm init` looks like this:
```kubeadm join --token 6c07ec.8afb08fc7a5d4f12 https://10.192.0.2:6443 --discovery-token-ca-cert-hash sha256:6d13f3d13a594b0ff84475df631aca33548a49334e691835b352189e8f9382d9```

while without it it's ```join --skip-preflight-checks --token 974143.35f0ff3e081ec3d5 10.192.0.2:6443 --discovery-token-ca-cert-hash sha256:67d376aaaa1338b35898f0f0ac5ef8359c8df8b458654e2942d87afaafab797f```

The `https://` version appears not to work, in particular, it broke kubeadm-dind-cluster for @pmichali  (but I think it will not work in non-DIND case, too)